scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -4010,39 +4126,46 @@ def gather_sample(self):
 
         # gather metrics
         containers = None
-        if self.__report_container_metrics and self.__client:
-            if (
-                self.__metrics_controlled_warmer is not None
-                and not self.__metrics_controlled_warmer.is_running()
-            ):
-                self.__metrics_controlled_warmer.set_k8s_cache(k8s_cache)
-                self.__metrics_controlled_warmer.start()
-            containers = _get_containers(
-                self.__client,
-                ignore_container=None,
-                glob_list=self.__glob_list,
-                k8s_cache=k8s_cache,
-                k8s_include_by_default=self.__include_all,
-                k8s_namespaces_to_include=self.__namespaces_to_include,
-                controlled_warmer=self.__metrics_controlled_warmer,
-            )
+        if self.__report_container_metrics:
+            if self.__client:
+                if (
+                    self.__metrics_controlled_warmer is not None
+                    and not self.__metrics_controlled_warmer.is_running()
+                ):
+                    self.__metrics_controlled_warmer.set_k8s_cache(k8s_cache)
+                    self.__metrics_controlled_warmer.start()
+                containers = _get_containers(
+                    self.__client,
+                    ignore_container=None,
+                    glob_list=self.__glob_list,
+                    k8s_cache=k8s_cache,
+                    k8s_include_by_default=self.__include_all,
+                    k8s_namespaces_to_include=self.__namespaces_to_include,
+                    controlled_warmer=self.__metrics_controlled_warmer,
+                )
+            else:
+                containers = self.__container_checker.get_containers()
         try:
             if containers:
+                stats = None
```

Comment:
I'm having some troubles understanding this piece of code which populates stats variable

Reply:
Yes, `__gather_k8s_metrics_from_kubelet` uses the /stats/summary results regardless of which runtime we are in, before my changes it always called that endpoint itself. In CRI we call that endpoint in my new code as well so I figure there was no reason calling the endpoint twice if we already had the result.

Reply:
Yes, `__gather_k8s_metrics_from_kubelet` uses the /stats/summary results regardless of which runtime we are in, before my changes it always called that endpoint itself. In CRI we call that endpoint in my new code as well so I figure there was no reason calling the endpoint twice if we already had the result.